docs/book/src/topics/workload-identity.md

```diff
@@ -0,0 +1,102 @@
+# Workload Identity
+
+Azure AD Workload identity is the next iteration of Azure AD Pod identity 
+that enables Kubernetes applications (e.g. CAPZ ) to access Azure cloud 
+resources securely with Azure Active Directory.
+
+This document describes a quick start guide of using workload identity and 
+assumes that you have access to Azure cloud.
+
+Workload identity is currently worked upon and cloud provider azure 
+integration is in progress. Please refer [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3589) issue for more details.
+For more details around it, please refer the [proposal](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/docs/proposals/20221611-workload-identity-integration.md)
+
+## Workload Identity Quick Start Guide
+
+- Create a public and private key pair. For example, you can generate the 
+  key pairs using OpenSSL.
+
+  Generate a private key called sa.key using the following command:
+```bash
+$openssl genrsa -out sa.key 2048
+```
+  Generate a public key using the private key. 
+```bash
+$openssl rsa -in sa.key -pubout -out sa.pub
+```
+
+- Create and upload Discovery and JWKS document using this [link](https://azure.github.io/azure-workload-identity/docs/installation/self-managed-clusters/oidc-issuer.html)
+
+- Create a federated identity credential using the steps outlined [here](https://azure.github.io/azure-workload-identity/docs/topics/federated-identity-credential.html)
+
+- Create a Kind cluster with necessary flags. You can use the following kind 
+  config yaml.
```

Comment:
I tried creating a kind cluster using this yaml file but got the following error. Not sure if it's something wrong on my end:
```
ERROR: failed to create cluster: could not determine kind / apiVersion for config: yaml: line 2: mapping values are not allowed in this context
```

Reply:
Good catch. I think it missed the `Kind: CLuster` line at the beginning. Let me do that. 
Plus, I think adding a command `kind create cluster --config config.yaml` should be nice.